src/containers/CreateCommunity.js

```diff
@@ -0,0 +1,434 @@
+
+  set = key => event => {
+    let { value } = event.target
+    this.setValue(key, value)
+    this.validate(key, value)
+  }
+
+  validate (key, value) {
+    switch (key) {
+      case 'name':
+        this.setError({nameBlank: !value})
+
+        if (!value) {
+          this.resetValidation('name')
+        } else {
+          return this.checkUnique('name', value)
+        }
+        break
+      case 'description':
+        this.setError({descriptionBlank: !value})
+        break
+      case 'slug':
+        let error = {slugBlank: false, slugInvalid: false}
+
+        if (!value) {
+          error.slugBlank = true
+        } else if (!value.match(/^[a-z0-9-+]+$/)) {
+          error.slugInvalid = true
+        }
+
+        this.setError(error)
+
+        if (some(error)) {
+          this.resetValidation('slug')
+        } else {
+          return this.checkUnique('slug', value)
+        }
+        break
+      case 'beta_access_code':
+        this.setError({codeBlank: !value})
+
+        if (!value) {
+          this.resetValidation('beta_access_code')
+        } else if (includes(value, '/')) {
+          this.setError({codeInvalid: true})
+        } else {
+          return this.checkUnique('beta_access_code', value)
+        }
+        break
+    }
+  }
+
+  attachImage (type) {
+    let { dispatch } = this.props
+    let community = {id: 'new', slug: 'new'}
+    switch (type) {
+      case 'avatar':
+        dispatch(uploadImage(avatarUploadSettings(community)))
+        break
+      case 'banner':
+        dispatch(uploadImage(bannerUploadSettings(community)))
+    }
+    return false
+  }
+
+  validateAll () {
+    let { community } = this.props
+    return Promise.all(
+      ['name', 'description', 'slug', 'beta_access_code', 'category']
+      .map(key => this.validate(key, community[key]))
+    )
+  }
+
+  submit = () => {
+    let { validating, dispatch, community, currentUser } = this.props
+    if (validating) return
+
+    this.validateAll().then(() => {
+      if (some(id => id, this.props.errors)) return scrollToBottom()
+
+      dispatch(createCommunity(community))
+      .then(({ payload }) => {
+        if (some(id => id, this.props.errors)) {
+          return scrollToBottom()
+        } else {
+          trackEvent(ADDED_COMMUNITY, {community})
+          saveCurrentCommunityId(dispatch, payload.community_id, currentUser.id)
+          dispatch(navigate(`/create/invite`))
+        }
+      })
+    })
+  }
+
+  render () {
+    const { community, errors } = this.props
+
+    const { expanded } = this.state
+
+    return <Modal title='Create your community.'
+      className='create-community-one'
+      subtitle="Let's get started unlocking the creative potential of your community with Hylo"
+      standalone>
+        <ModalInput label='Name' ref='name' onChange={this.set('name')}/>
+        {errors.nameBlank && <p className='help error'>Please fill in this field.</p>}
```

Comment:
these errors show up in weird places -- do you think we could tuck them into the ModalInput and right-justify them on the same line as the label? you could "poke a hole" into the component by adding an attribute and moving these snippets into a `<div>` inside them, like we do with Dropdown and toggleChildren.